features/topology-map/plugins/org.opennms.features.topology.plugins.topo.graphml/src/main/java/org/opennms/features/topology/plugins/topo/graphml/GraphMLSearchProvider.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2016 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2016 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.topology.plugins.topo.graphml;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.opennms.features.topology.api.GraphContainer;
+import org.opennms.features.topology.api.topo.SearchQuery;
+import org.opennms.features.topology.api.topo.SimpleSearchProvider;
+import org.opennms.features.topology.api.topo.VertexRef;
+import org.opennms.features.topology.plugins.topo.graphml.model.GraphML;
+import org.opennms.features.topology.plugins.topo.graphml.model.GraphMLNode;
+
+public class GraphMLSearchProvider extends SimpleSearchProvider {
+
+    private final GraphMLTopologyProvider graphMLTopologyProvider;
+
+    public GraphMLSearchProvider(GraphMLTopologyProvider graphMLTopologyProvider) {
+        this.graphMLTopologyProvider = Objects.requireNonNull(graphMLTopologyProvider);
+    }
+
+    @Override
+    public String getSearchProviderNamespace() {
+        return graphMLTopologyProvider.getVertexNamespace();
+    }
+
+    @Override
+    public List<? extends VertexRef> queryVertices(SearchQuery searchQuery, GraphContainer container) {
+        final List<GraphMLVertex> matchingVertices = new ArrayList<>();
+        final GraphML graphML = graphMLTopologyProvider.getGraphML();
```

Comment:
I think we should use `graphMLTopologyProvider.getVertices()` and filter on the result instead.
But we can also leave it for now, as this is going to change anyways as soon as we have the layering in place.